### PR TITLE
storage: rename seek to scan (phase 1).

### PR DESCRIPTION
**Pull request by ngaut**
I will change definition in pb later (phase 2)

### CODE REVIEW

File: src/storage/engine/memory.rs [hunk 1]
```diff
@@ -44,8 +44,8 @@ impl Engine for EngineBtree {
         Ok(m.get(key.raw()).cloned())
     }
 
-    fn seek(&self, _: &Context, key: &Key) -> Result<Option<KvPair>> {
-        trace!("EngineBtree: seek {}", key);
+    fn scan(&self, _: &Context, key: &Key) -> Result<Option<KvPair>> {
```
Comment (BusyJay) [inline src/storage/engine/memory.rs]: I think scan should return a Vec.

Comment (ngaut) [inline src/storage/engine/memory.rs]: That should be in later PR. Let's finish renaming first.

Comment (BusyJay) [inline src/storage/engine/memory.rs]: Why not just do it in one pr? It looks weird.

Comment (coocood) [inline src/storage/engine/memory.rs]: I think `scan` need `startKey`, `endKey` and `limit`, and returns a list of KvPair.
If we need scan we better add a method `scan` along with `seek`.

Comment (ngaut) [inline src/storage/engine/memory.rs]: Because we need to change multiple repo.

**Pull request closed without merging**